src/app/core/export/export-service/export.service.ts

```diff
@@ -34,34 +35,100 @@ export class ExportService {
    */
-  createCsv(data: any[], config?: ExportColumnConfig[]): string {
-    const allFields = new Set();
+  async createCsv(data: any[], config?: ExportColumnConfig[]): Promise<string> {
     const exportableData = [];
+    const columns = new Set<string>(
+      (config ?? []).map((c) => c.label ?? c.query)
+    );
 
-    data.forEach((element: any) => {
-      const exportableObj = {};
-
-      const currentRowConfig =
+    for (const dataRow of data) {
+      const rowConfig =
         config ??
-        Object.keys(element).map((key) => ({ key: key } as ExportColumnConfig));
-      for (const columnConfig of currentRowConfig) {
-        const label = columnConfig.label ?? columnConfig.key;
-        const value = entityListSortingAccessor(element, columnConfig.key);
-        if (value?.toString().match(/\[object.*\]/) !== null) {
-          // skip object values that cannot be converted to a meaningful string
-          continue;
-        }
-
-        exportableObj[label] = value;
-        allFields.add(label);
+        Object.keys(dataRow).map(
+          (key) => ({ query: key } as ExportColumnConfig)
+        );
+      if (!config) {
+        Object.keys(dataRow).forEach((k) => columns.add(k));
       }
 
-      exportableData.push(exportableObj);
-    });
+      const exportableElement = await this.transformDataRowToExportRow(
+        dataRow,
+        rowConfig
+      );
+
+      const extendedExportableRows = this.extendIntoMultipleRows(
+        exportableElement,
+        rowConfig
+      );
+      exportableData.push(...extendedExportableRows);
+    }
 
     return this.papa.unparse(
-      { data: exportableData, fields: [...allFields] },
-      { quotes: true, header: true }
+      { data: exportableData, fields: Array.from(columns.values()) },
+      { quotes: true, header: true, newline: ExportService.SEPARATOR_ROW }
+    );
+  }
+
+  private async transformDataRowToExportRow(
+    element: any,
+    config: ExportColumnConfig[]
+  ) {
+    const exportableObj = {};
+
+    for (const columnConfig of config) {
+      const label = columnConfig.label ?? columnConfig.query;
+      let value;
+      if (isQuery(columnConfig.query)) {
+        value = await this.queryService.queryData(
+          columnConfig.query,
+          null,
+          null,
+          [element]
+        );
+      } else {
+        value = entityListSortingAccessor(element, columnConfig.query);
+      }
+
+      if (value?.toString().match(/\[object.*\]/) !== null) {
+        // skip object values that cannot be converted to a meaningful string
+        continue;
+      }
+
+      exportableObj[label] = value;
+    }
+
+    return exportableObj;
+
+    function isQuery(queryKey) {
+      return queryKey.startsWith(".") || queryKey.query.startsWith(":");
+    }
+  }
+
+  private extendIntoMultipleRows(
+    exportableElement: {},
+    rowsToExtend: ExportColumnConfig[]
+  ): any[] {
+    const multiElementColumns = rowsToExtend.filter(
```

Comment:
I think this should be moved outside of this function because it only makes sense on the first step of the recursion